test-cases/trasferred-value/lib.rs

```diff
@@ -0,0 +1,124 @@
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+#[ink::contract]
+mod transferred_value {
+
+    #[ink(storage)]
+    pub struct TransferredValue {}
+
+    impl TransferredValue {
+        /// Creates a new Template contract.
+        #[ink(constructor, payable)]
+        pub fn new() -> Self {
+            Self {}
+        }
+
+        #[ink(message, payable)]
+        pub fn send_tokens_and_get_transferred_message(&self) -> Balance {
+            self.env().transferred_value()
+        }
+
+        #[ink(message, payable)]
+        pub fn was_it_one_hundred(&self) {
+            ink::env::debug_println!("received payment: {}", self.env().transferred_value());
+            assert!(
+                self.env().transferred_value() == 100,
+                "payment was not one hundred"
+            );
+        }
+    }
+
+    #[cfg(test)]
+    mod tests {
+        use ink::env::{
+            test::{
+                callee, default_accounts, get_account_balance, set_account_balance, set_caller,
+            },
+            DefaultEnvironment,
+        };
+
+        use super::*;
+
+        #[ink::test]
+        fn transferred_value() {
+            use ink::codegen::Env;
+            let contract = create_contract();
+            let accounts = default_accounts::<DefaultEnvironment>();
+            let contract_account = contract.env().account_id();
+
+            set_account_balance::<DefaultEnvironment>(accounts.eve, 1000);
+            set_account_balance::<DefaultEnvironment>(contract_account, 0);
+            set_caller::<DefaultEnvironment>(accounts.eve);
+
+            let res =
+                ink::env::pay_with_call!(contract.send_tokens_and_get_transferred_message(), 1000);
+
+            let contract_new_balance = get_balance(contract_account);
+            let caller_new_balance = get_balance(accounts.eve);
+
+            assert_eq!(caller_new_balance, 0);
+            assert_eq!(contract_new_balance, 1000);
+            assert_eq!(res, 1000);
+        }
```

Comment:
Test is not following the `given`, `when`, `then` pattern. Please refer to the other test-cases.